acceptance_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clickhouse
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sync/atomic"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/config"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/matryer/is"
+)
+
+type driver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+
+	counter int32
+}
+
+// GenerateRecord generates a random sdk.Record.
+func (d *driver) GenerateRecord(t *testing.T, operation sdk.Operation) sdk.Record {
+	atomic.AddInt32(&d.counter, 1)
+
+	return sdk.Record{
+		Position:  nil,
+		Operation: operation,
+		Metadata: map[string]string{
+			"clickhouse.table": d.Config.SourceConfig[config.Table],
+		},
+		Key: sdk.StructuredData{
+			"int_type": d.counter,
+		},
+		Payload: sdk.Change{After: sdk.RawData(
+			fmt.Sprintf(`{"int_type":%d,"string_type":"%s"}`, d.counter, uuid.NewString()),
+		)},
+	}
+}
+
+func TestAcceptance(t *testing.T) {
+	cfg := prepareConfig(t)
+
+	is := is.New(t)
+
+	sdk.AcceptanceTest(t, &driver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest: func(t *testing.T) {
+					err := createTable(cfg[config.URL], cfg[config.Table])
+					is.NoErr(err)
+				},
```

Comment:
Hm. 
IIUC, the `AfterTest` func calls deferring.
Also, it's no sense to drop the table in the `BeforeTest` func, because it will be a new name of a table there (the table name creates in a `prepareConfig` function).
Am I wrong?

Reply:
Yes, AfterTest is called after deferring but if the test is killed, it may not come to that. However, given that you always create a new name, and that this typically runs in Docker, my suggestion doesn't make a lot of sense.:)